File: lib/commands/today-command.js
'use strict';

const BaseCommand = require('./base-command');
const moment = require('moment');
const Menu = require('../../data/models/menu');

class TodayCommand extends BaseCommand {
  handle() {
    return new Promise((resolve, reject) => {
      Menu.findOne({date: moment().format('YYYYMMDD')}, (error, result) => {
        if (error)
          resolve('Error getting menu data.');
        else {
          if(!result)
          {
            resolve('I don\'t know which meals are being served today.!');
            return;
          }
          let message = `Today is the *${moment().format('DD.MM.YYYY')}* and the meals are:\n`;
          result.meals.forEach(meal => {
            message += `  ${meal}\n`;
          });
          resolve(message);
        }
      });
    });
  }
}

module.exports = TodayCommand;

File: src/Deathstar.Canteen/Commands/TodayCommand.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Deathstar.Canteen.Commands.Abstractions;
using Deathstar.Canteen.Persistence;
using Deathstar.Canteen.Slack;

namespace Deathstar.Canteen.Commands
{
	public class TodayCommand : ICommand
	{
		private readonly IMenuCollection menuCollection;
		private readonly ISlackbot slackbot;

		public TodayCommand(IMenuCollection menuCollection, ISlackbot slackbot)
		{
			this.menuCollection = menuCollection;
			this.slackbot = slackbot;
		}

		public string HelpText { get; } = "The *today* command will return a list of today's meals.";

		public string Name { get; } = "today";

		public async Task HandleAsync(ICommandMessage message, CancellationToken cancellationToken)
		{
			DateTime date = DateTime.Today;
			Menu menu = await menuCollection.SingleOrDefaultAsync(x => x.Date == date.ToString("yyyyMMdd"), cancellationToken);
			string response = menu == null
				? "I don't know which meals are being served today!"
				: $"Today is the *{date:dd.MM.yyyy}* and the meals are:{Environment.NewLine}{menu}";

			slackbot.SendMessage(message.Channel, response);
		}
	}
}
